habitat-lab/habitat/tasks/rearrange/multi_task/domain_configs/fp.yaml

```diff
@@ -14,56 +14,66 @@ constants: {}
 
 
 predicates:
-  - name: in
-    args:
-      - name: obj
-        expr_type: obj_type
-      - name: receptacle
-        expr_type: art_receptacle_entity_type
-    set_state:
-      obj_states:
-        obj: receptacle
-
   - name: holding
     args:
       - name: obj
         expr_type: movable_entity_type
-      - name: robot_id
+      - name: robot
         expr_type: robot_entity_type
-    set_state:
-      robot_states:
-        robot_id:
-          holding: obj
+    is_valid_fn:
+      # Function that checks if the predicate is satisfied. See
+      # `pddl_defined_predicates.py` for the pre-defined instances of these
+      # functions.
+      _target_: habitat.tasks.rearrange.multi_task.pddl_defined_predicates.is_robot_hold_match
+      hold_state: True
+    set_state_fn:
+      # Funtion that sets simulator state based on predicate arguments.
+      _target_: habitat.tasks.rearrange.multi_task.pddl_defined_predicates.set_robot_holding
+      hold_state: True
+
+  - name: in
+    args:
+      - name: obj
+        expr_type: obj_type
+      - name: recep
+        expr_type: art_receptacle_entity_type
+    is_valid_fn:
+      _target_: habitat.tasks.rearrange.multi_task.pddl_defined_predicates.is_inside
 
   - name: not_holding
     args:
-      - name: robot_id
+      - name: robot
         expr_type: robot_entity_type
-    set_state:
-      robot_states:
-        robot_id:
-          should_drop: True
-
+    is_valid_fn:
+      _target_: habitat.tasks.rearrange.multi_task.pddl_defined_predicates.is_robot_hold_match
+      hold_state: False
+    set_state_fn:
+      _target_: habitat.tasks.rearrange.multi_task.pddl_defined_predicates.set_robot_holding
+      hold_state: False
   - name: robot_at
     args:
-      - name: Y
+      - name: at_entity
         expr_type: static_obj_type
-      - name: robot_id
+      - name: robot
         expr_type: robot_entity_type
-    set_state:
-      robot_states:
-        robot_id:
-          pos: Y
+    is_valid_fn:
+      _target_: habitat.tasks.rearrange.multi_task.pddl_defined_predicates.is_robot_at_position
+      dist_thresh: 2.0
+    set_state_fn:
+      _target_: habitat.tasks.rearrange.multi_task.pddl_defined_predicates.set_robot_position
+      dist_thresh: 2.0
 
   - name: at
```

Comment:
might name this object_at to be consistent with robot_at

Reply:
This required refactoring a lot more files, but this is added now.